core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/InstanceGroupNetworkV1ToInstanceGroupNetworkV4Converter.java

```diff
@@ -0,0 +1,194 @@
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.parameter.instancegroup.network.aws.InstanceGroupAwsNetworkV4Parameters;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.parameter.instancegroup.network.azure.InstanceGroupAzureNetworkV4Parameters;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.parameter.instancegroup.network.gcp.InstanceGroupGcpNetworkV4Parameters;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.parameter.instancegroup.network.mock.InstanceGroupMockNetworkV4Parameters;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.parameter.instancegroup.network.openstack.InstanceGroupOpenstackNetworkV4Parameters;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.instancegroup.network.InstanceGroupNetworkV4Request;
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.distrox.api.v1.distrox.model.network.InstanceGroupNetworkV1Request;
+import com.sequenceiq.distrox.api.v1.distrox.model.network.aws.InstanceGroupAwsNetworkV1Parameters;
+import com.sequenceiq.distrox.api.v1.distrox.model.network.azure.InstanceGroupAzureNetworkV1Parameters;
+import com.sequenceiq.distrox.api.v1.distrox.model.network.gcp.InstanceGroupGcpNetworkV1Parameters;
+import com.sequenceiq.distrox.api.v1.distrox.model.network.mock.InstanceGroupMockNetworkV1Parameters;
+import com.sequenceiq.distrox.api.v1.distrox.model.network.openstack.InstanceGroupOpenstackNetworkV1Parameters;
+import com.sequenceiq.environment.api.v1.environment.model.response.DetailedEnvironmentResponse;
+import com.sequenceiq.environment.api.v1.environment.model.response.EnvironmentNetworkResponse;
+
+@Component
+public class InstanceGroupNetworkV1ToInstanceGroupNetworkV4Converter {
+
+    public InstanceGroupNetworkV4Request convertToInstanceGroupNetworkV4Request(Pair<InstanceGroupNetworkV1Request, DetailedEnvironmentResponse> network) {
+        DetailedEnvironmentResponse value = network.getValue();
+        EnvironmentNetworkResponse environmentNetworkResponse = null;
+        if (value == null) {
+            environmentNetworkResponse = new EnvironmentNetworkResponse();
+        } else {
+            environmentNetworkResponse = value.getNetwork();
+        }
+        InstanceGroupNetworkV1Request key = network.getKey();
+        if (key == null) {
+            key = new InstanceGroupNetworkV1Request();
+        }
+
+        InstanceGroupNetworkV4Request request = new InstanceGroupNetworkV4Request();
+
+        if (network.getValue() != null) {
+            switch (value.getCloudPlatform()) {
+                case "AWS":
+                    request.setAws(getAwsNetworkParameters(Optional.ofNullable(key.getAws()), environmentNetworkResponse));
```

Comment:
why don't you use the `ProviderParameterCalculator providerParameterCalculator;`?